src/views/dropdown.rs

```diff
@@ -30,28 +31,120 @@ prop_extractor!(DropdownStyle {
     close_on_accept: CloseOnAccept,
 });
 
-pub fn dropdown<T, MF, I, LF, AIF>(
-    active_item: AIF,
-    main_view: MF,
-    iterator: I,
-    list_item_fn: LF,
-) -> Dropdown<T>
-where
-    MF: Fn(T) -> Box<dyn View> + 'static,
-    I: IntoIterator<Item = T> + Clone + 'static,
-    LF: Fn(T) -> Box<dyn View> + Clone + 'static,
-    T: Clone + 'static,
-    AIF: Fn() -> T + 'static,
-{
-    Dropdown::new(active_item, main_view, iterator, list_item_fn)
-}
-
-/// A dropdown widget
+/// Create a dropdown widget.
+///
+/// # Example
+///
+/// A common scenario is populating a dropdown menu using an enum.
+///
+/// The below example creates a dropdown with three items, one for each character in our
+/// `Character` enum.
+///
+/// The `strum` crate can be useful. In this example, we use `strum` to create an iterator for our
+/// `Character` enum.
+///
+/// First, define the enum:
+/// ```rust
+/// use strum::IntoEnumIterator;
+///
+/// #[derive(Clone, strum::EnumIter)]
+/// enum Character {
+///     Ori,
+///     Naru,
+///     Gumo,
+/// }
+///
+/// // Implement Display so we can call .to_string() lower down.
+/// impl std::fmt::Display for Character {
+///     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+///         match self {
+///             Self::Ori => write!(f, "Ori"),
+///             Self::Naru => write!(f, "Naru"),
+///             Self::Gumo => write!(f, "Gumo"),
+///         }
+///     }
+/// }
+/// ```
+///
+/// Then, create the dropdown:
+///
+/// ```rust
+/// # use strum::IntoEnumIterator;
+/// # #[derive(Clone, strum::EnumIter)]
+/// # enum Character {
+/// #     Ori,
+/// #     Naru,
+/// #     Gumo,
+/// # }
+/// # // Implement Display so we can call .to_string() lower down.
+/// # impl std::fmt::Display for Character {
+/// #     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+/// #         match self {
+/// #             Self::Ori => write!(f, "Ori"),
+/// #             Self::Naru => write!(f, "Naru"),
+/// #             Self::Gumo => write!(f, "Gumo"),
+/// #         }
+/// #     }
+/// # }
+/// #
+/// # fn character_select() -> impl floem::IntoView {
+/// use floem::{reactive::*, views::*, IntoView};
+///
+/// let mode = RwSignal::new(Character::Ori);
+///
+/// dropdown::dropdown(
+///     move || mode.get(),
+///     move |m| text(m.to_string()).into_any(),
+///     Character::iter(),
+///     move |m| text(m.to_string()).into_any(),
+/// )
+/// .on_accept(move |new| mode.set(new))
+/// # }
+/// ```
+///
+/// You can display the selected item differently than the list of items.
```

Comment:
I think just the single example showing how to build and use the enum iter is sufficient.